src/lib/util/corestr.cpp

```diff
@@ -222,7 +222,7 @@ std::string strmakeupper(std::string& str)
  * @return  A std::string.
```

Comment:
Could you make this doxy comment a bit more useful?

The `@brief` description just "verbises" the function name - if you don't know what it does, this isn't going to help.

For the `@return` entry, firstly it returns a reference to a `std::string`, secondly the comment should not just repeat what one can see in the declaration, and thirdly it should hint at the significance of the returned string reference.